content/en/core/overview/self-hosting-multiple-nodes.md

```diff
@@ -0,0 +1,285 @@
+To add a worker to this swarm, run the following command:
+
+    docker swarm join --token <very-long-token-value> <main-server-private-ip>:2377
+
+To add a manager to this swarm, run 'docker swarm join-token manager' and follow the instructions.
+
+```
+
+2. Create overlay network
+
+`sudo docker network create --driver=overlay --attachable cht-overlay`
+
+### Node2, Node3 & Node4
+
+On each of these nodes run the join command given to you above.
+
+`docker swarm join --token <very-long-token-value> <main-server-private-ip>:2377`
+
+
+## Configuration - Node 1 - CHT-core
+
+### Setup Docker
+
+1. `sudo apt update`
+2. `sudo apt install docker.io`
+3. `sudo apt install docker-compose`
+
+### Directory Structure
+
+Create the following directory structure. This is important.
+
+```
+|--CHT
+    |__ compose/
+    |__ certs/
+    |__ upgrade-service/
+```
+
+Explanations:
+
+* compose - contains the docker-compose files for cht-core and couchdb.
+* certs - the path that you'll put SSL certificates.
+* upgrade-service - the path for the docker-compose file of the upgrade-service.
+
+### Download required docker-compose files
+
+The following commands download the 4.0.1 version. If you want a different version, you would have to change `medic:4.0.1` to the version you'd like to install.
+
+1. Download the cht-core docker-compose file. Run the following command from the `CHT` directory above.
+`curl -s -o ./compose/cht-core.yml https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:4.0.1/docker-compose/cht-core.yml`
+2. Download the upgrade service docker-compose file.
+`curl -s -o ./upgrade-service/docker-compose.yml https://raw.githubusercontent.com/medic/cht-upgrade-service/main/docker-compose.yml`
+
+### Docker-compose modifications
+
+1. Go to the networks section and add the following:
+
+```
+  cht-overlay:
+    driver: overlay
+    external: true
+```
+
+2. Go to each of the containers and add the following:
+
+```
+    networks:
+      - cht-overlay
+```
+
+### Prepare Environment Variables file
+
+Prepare a `.env` file that contains the following variables and save it in the upgrade-service directory. Note that some of the values are not necessary in this node but for the purpose of making things simple here, we'll create one env file and use it on all nodes.  We are assuming that your home directory is `/home/ubuntu/`. If that's not accurate, please change that in the values below
+
+```
+CHT_COMPOSE_PROJECT_NAME=cht
+COUCHDB_SECRET=<some-super-long-combination-of-alphanumeric-characters>
+DOCKER_CONFIG_PATH=/home/ubuntu/cht/upgrade-service
+COUCHDB_DATA=/home/ubuntu/cht/couchdb
+CHT_COMPOSE_PATH=/home/ubuntu/cht/compose
+COUCHDB_USER=medic
+COUCHDB_PASSWORD=<another-super-long-alphanumeric-characters>
+CERTIFICATE_MODE=OWN_CERT
+COUCHDB_UUID=<put-in-a-guid-value>
+SSL_VOLUME_MOUNT_PATH=/etc/nginx/private/
+SSL_CERT_FILE_PATH=/etc/nginx/private/cert.pem
+SSL_KEY_FILE_PATH=/etc/nginx/private/key.pem
```

Comment:
These are the same as the default values in the compose file so we can remove them to simplify the `.env` file

Reply:
done in 946fb8c